index.d.ts

```diff
@@ -878,14 +878,11 @@ type ParseTReturn<Key, Res> = Key extends `${infer K1}${_KeySeparator}${infer Re
 
 type TReturnOptionalNull = _ReturnNull extends true ? null : never;
 type TReturnOptionalObjects<TOpt extends TOptions> = _ReturnObjects extends true
-  ? $SpecialObject
+  ? $SpecialObject | string
```

Comment:
This makes the same problem when _ReturnObjects is true.

Reply:
Check my comment: https://github.com/i18next/i18next/issues/1982#issuecomment-1603288015

Reply:
`_ReturnObjects` is out of question. Users can access the actual value (string) if they pass the full `key`

Reply:
Then this pr cannot close the issue.

Reply:
Actually, there seems to be no reason to change here.

Reply:
It's not because the `returnObjects` is set to `true` in the init options (`i18next.init({returnObjects: true})`), that `t` function will return an object. if the full key is passed, it'll return a string. That's why `string | $SpecialObject` is accurate here. For example:

```tsx
const translation = {
  en: {
    namespace: {
      foo: {
        bar: ''
      }
    }
  }
}

i18next.init({
  resources: translations,
  returnObjects: true
})

// return an object
i18next.t('namespace:foo');

//return an string
i18next.t('namespace:foo.bar');
```

Reply:
Your example doesn't seem to set _ReturnObjects true.

Reply:
Answer updated!

Reply:
Still got an error. Why you thought that returned a string?

```ts
// Property 'trim' does not exist on type 'DefaultTReturn<TOpt>'.
//   Property 'trim' does not exist on type 'object'.ts(2339)
i18next.t('namespace:foo.bar').trim()
```

Reply:
You need to narrow it down to ensure that's not an object, but a string.
```tsx
const result = i18next.t('namespace:foo.bar');
if (typeof result === 'string') {
  result.trim();
}
```

Reply:
Then your comment `return a string` is a lie. The problem is not fixed.

Reply:
Based on your assumptions and answers, it seems like you might be lacking some knowledge and making incorrect assumptions about how Typescript works. Before making groundless opinions and wrong assumptions, I suggest you take some time to read TypeScript documentation or tutorials to get your facts straight.

Reply:
Self-conceit answer. You are writing the wrong test due to your lack of even rudimentary knowledge that the difference of type casts and type annotations.

https://github.com/i18next/i18next/blob/5fb74d516ccd6bfee4a6a3e5339ef1bdd52c55ee/test/typescript/t.test.ts#L16-L22

And you don't even understand the types you wrote.

>@[falsandtru](https://github.com/falsandtru) falsandtru [1 hour ago](https://github.com/i18next/i18next/pull/1985#discussion_r1239127245)
Your example doesn't seem to set _ReturnObjects true.

>@[pedrodurek](https://github.com/pedrodurek) pedrodurek [48 minutes ago](https://github.com/i18next/i18next/pull/1985#discussion_r1239130847)
Answer updated!

***You don't even understand the types you wrote.***

Reply:
I'm tired of reading your code full of mistakes. Do commit the mistake-ridden code as before.